osf/models/user.py

```diff
@@ -709,10 +711,14 @@ def merge_user(self, user):
                 new_name_format = '{} ({}){}'
                 new_name = new_name_format.format(name_without_extension, digit, extension)
 
-                # check if new name conflicts, update til it does not
-                while BaseFileNode.objects.filter(node=primary_quickfiles, name=new_name, type='osf.osfstoragefile').exists():
+                # check if new name conflicts, update til it does not (try up to 1000 times)
+                rename_count = 0
+                while OsfStorageFileNode.objects.filter(node=primary_quickfiles, name=new_name).exists():
                     digit += 1
                     new_name = new_name_format.format(name_without_extension, digit, extension)
+                    rename_count += 1
+                    if rename_count >= MAX_QUICKFILES_MERGE_RENAME_ATTEMPTS:
+                        raise ValueError('Maximum number of rename attempts has been reached')
```

Comment:
AH that's perfect! I'll do that. I was looking for a better error, put Value as kind of a placeholder and then forgot to change it...